@here/harp-materials/lib/EdgeMaterial.ts

```diff
@@ -19,12 +19,17 @@ import { enforceBlending, setShaderDefine, setShaderMaterialDefine } from "./Uti
 const vertexSource: string = `
 #define EDGE_DEPTH_OFFSET 0.0001
 
-attribute vec3 position;
 attribute vec4 color;
 
+// SHADER_NAME may be defined by THREE.JS own shaders in which case these attributes & uniforms are
```

Comment:
the class `EdgeMaterial` extends `THREE.RawShaderMaterial`, which isn't a THREE.js shader, so my guess is that this will be never defined?

Reply:
There are cases when it will be defined, which then leads to the following runtime error:

three.min.js:83 THREE.WebGLProgram: shader error:  0 35715 false gl.getProgramInfoLog No compiled vertex shader when at least one graphics shader is attached.
 THREE.WebGLShader: gl.getShaderInfoLog() vertex
ERROR: 0:54: 'position' : redefinition
ERROR: 0:57: 'modelViewMatrix' : redefinition
ERROR: 0:58: 'projectionMatrix' : redefinition
ERROR: 0:59: 'normalMatrix' : redefinition
1: precision highp float;
2: precision highp int;
3: #define HIGH_PRECISION
4: #define SHADER_NAME ShaderMaterial
...

Reply:
This is because the material is being reconstructed in Outline.ts, I see now. Ok, all good.